Worked on the menu and problem selector.

paulaOliverosHollis opened the pull request:
Program - is now working. You can select and see problems and solutions. I made sure to add several validation methods but I have yet to add any exception catching and handling.

Problems - changed the accessibility to public and same a very small change in formatting.

DynamicProblemSelector- I made a very tiny change in wording.

paulaOliverosHollis committed:
Worked on the menu and problem selector.

Program - is now working. You can select and see problems and solutions. I made sure to add several valification methods but I have yet to add any exception catching and handeling.

zhollis21 commented:
Rejected. This is crap

On Wed, Sep 25, 2019 at 2:54 PM Paula Oliveros Hollis <
notifications@github.com> wrote:

> @paulaOliverosHollis <https://github.com/paulaOliverosHollis> requested
> your review on: #2
> <https://github.com/paulaOliverosHollis/Project-Euler/pull/2> Worked on
> the menu and problem selector..
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/paulaOliverosHollis/Project-Euler/pull/2?email_source=notifications&email_token=AKOEW53IJKLB3EP6LOTDJP3QLOXVPA5CNFSM4I2QX4QKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOT23MZCI#event-2662780041>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKOEW544BXREVRZSKENXYDTQLOXVPANCNFSM4I2QX4QA>
> .
>
-- 

     Thanks,
          Zach Hollis

zhollis21 commented on Project-Euler/Project-Euler/DynamicProblemSelector.cs:
```diff
@@ -18,7 +18,7 @@ public static void PrintProblem(int problemNumber, Type objectType)
             else
             {
-                Console.WriteLine($"Problem {problemNumber} is not a valid selection. Please try again!");
+                Console.WriteLine($"Sorry, Problem {problemNumber} is not available. Please try another problem!");
```
I like it!

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -11,7 +11,24 @@ class Program
+            {
+                int userSelection = GetMenuSelection();
+
+                if (userSelection == 0)
```
You should use an enum instead of "magic numbers"

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -11,7 +11,24 @@ class Program
+                {
+                    DynamicProblemSelector.PrintAllProblems();
+                }
+                else if (userSelection == 1)
```
Same, this isn't very readable or maintainable.

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -11,7 +11,24 @@ class Program
+                {
+                    DynamicProblemSelector.PrintProblem(GetProblemNumber(), typeof(Problems));
+                }
+                else
```
```suggestion
                else // User Chose to Quit
```

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -20,19 +37,61 @@ public static void PrintIntro()
+
+            while (true)
+            {
+                string userSelection = Console.ReadLine().ToUpper();
```
If the user ever just hits <Enter> without typing anything, I think _Console.ReadLine()_ will return null. Which will blow up on _.ToUpper()_. To fix this add a question mark before _.ToUpper();_ (Google what a question mark before a method call will do)

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -20,19 +37,61 @@ public static void PrintIntro()
+            {
+                string userSelection = Console.ReadLine().ToUpper();
+
+                if (!string.IsNullOrEmpty(userSelection) && userSelection.Length == 1)
```
I would use _string.IsNullOrWhiteSpace_ as that will check for

- Null
- Empty
- and Whitespace

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -20,19 +37,61 @@ public static void PrintIntro()
+
+                    if (menuSelection == 'A')
+                    {
+                        return 0;
```
Again, I would use an Enum instead of passing around magic numbers

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -20,19 +37,61 @@ public static void PrintIntro()
+                    }
+                }
+                else
+                {
```
I would remove the _else_ because you want this code to execute even if it made it into the above if statement, but maybe they typed 'C' which wasn't valid.

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -20,19 +37,61 @@ public static void PrintIntro()
-        public static void PrintMenuOptions() //This method will be private once I write the method that will call it. blah blah
+        public static int GetMenuSelection()
+        {
+            PrintMenuOptions();
```
Instead of putting _PrintMenuOptions_ in 2 places (here and at the bottom of the while loop), just put it at the top of the while loop

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -20,19 +37,61 @@ public static void PrintIntro()
             Console.WriteLine("\n\n\tSelect One Of The Following Options:\n" +
                               "\n\tA - See All Problems and Solutions" +
-                              "\n\tB - See One Specific Problem and Solution" +
+                              "\n\tS - See One Specific Problem and Solution" +
```
Zach Approved Char :+1:

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -20,19 +37,61 @@ public static void PrintIntro()
         }
 
-        private static bool IsMenuSelectionValid()
+        private static int GetProblemNumber()
```
Looks good!

zhollis21 reviewed (changes requested):
Good work overall, only needs a couple of small changes.

paulaOliverosHollis committed:
Update Project-Euler/Project-Euler/Program.cs

Co-Authored-By: zhollis21 <43797367+zhollis21@users.noreply.github.com>

paulaOliverosHollis committed:
Created an enum with the menu options.

Added a ? in front of ToUpper to make sure the program would not crash if null. Also added a try catch.

Fixed some minor issues from pull request.

paulaOliverosHollis committed:
Merge branch 'WorkingOnMenuAndPloblemSelector' of https://github.com/paulaOliverosHollis/Project-Euler into WorkingOnMenuAndPloblemSelector

paulaOliverosHollis commented:
I have fixed the issues you requested. They are in my latest commit.

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -11,28 +11,93 @@ class Program
+            }
         }
 
+        public enum MenuOptions { A, S, Q };
```
Three suggestions:

- Can we make these Enums more descriptive? i.e. AllProblems, SelectProblem, Quit
- Since this enum will only ever be used in this class, it should be private.
- Put the enum in it's own space not hidden inbetween methods

zhollis21 commented on Project-Euler/Project-Euler/Program.cs:
```diff
@@ -11,28 +11,93 @@ class Program
+                PrintMenuOptions();
+
+                try
+                {
```
Why did you put this in a try block? 

1. I don't think this is necessary
2. It's very bad practice to use a try with an empty catch, you are basically just ignoring errors

Google about empty catch blocks and why they are bad and also when is the best time to use a try-catch

paulaOliverosHollis committed:
Removed Try Catch.

I also change the enum names to something more descriptive.

zhollis21 reviewed (approved):
Looks great! Nice job!
